crates/oxc_sourcemap/src/encode.rs

```diff
@@ -238,6 +286,21 @@ impl<'a> PreAllocatedString<'a> {
     }
 }
 
+#[inline]
+fn needs_escaping(s: &str) -> bool {
+    let bytes = s.as_bytes();
+
+    // Check for control characters (0x00-0x1F)
+    if memchr_iter(b'\x00', bytes).next().is_some() {
```

Comment:
Doesn't this only check for bytes which are 0, rather than all control chars (0x00-0x1F)?